subcommands/keys/ca_renewal_resign_device_ca.go

```diff
@@ -0,0 +1,66 @@
+package keys
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	//"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/foundriesio/fioctl/x509"
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "re-sign-device-ca <PKI Directory> [<Old PKI Directory>]",
+		Short: "Re-sign all existing Device CAs with a new root CA for your Factory PKI",
+		Run:   doReSignDeviceCaRenewal,
+		Args:  cobra.RangeArgs(1, 2),
+		Long: `Re-sign all existing Device CAs with a new root CA for your Factory PKI.
+
+Both currently active and disabled Device CAs are being re-signed.
+All their properties are preserved, including a serial number.
+Only the signature and authority key ID are being changed.
+This allows old certificates (issued by a previous root CA) to continue being used to issue device client certificates.
+
+Re-signed device CA certificates are stored in the provided PKI directory.
+An old PKI directory is used to locate corresponding private keys, and copy them into the PKI directory.
```

Comment:
shouldn't we use what we know from the server `fioctl keys ca show --just-device-cas`. Trusting this local directory is correct seems error-prone?

Reply:
We need to copy private keys from an old PKI directory to the new PKI directory. Otherwise, a user might lose them when deleting an old directory.

We don't have those private keys on our servers. But, we have their public parts which we can use for a match.